src/blockstorage/blockstorage.cpp

```diff
@@ -605,9 +605,9 @@ bool FlushStateToDiskInternal(CValidationState &state,
     bool fPeriodicFlush =
         mode == FLUSH_STATE_PERIODIC && nNow > nLastFlush + (int64_t)DATABASE_FLUSH_INTERVAL * 1000000;
     // Combine all conditions that result in a full cache flush.
-    bool fDoFullFlush = (mode == FLUSH_STATE_ALWAYS) || fCacheCritical || fPeriodicFlush || fFlushForPrune;
+    bool fDoFullFlush = (mode == FLUSH_STATE_ALWAYS) || fCacheCritical || fPeriodicFlush;
     // Write blocks and block index to disk.
-    if (fDoFullFlush || fPeriodicWrite)
+    if (fDoFullFlush || fPeriodicWrite || fFlushForPrune)
```

Comment:
So we decouple `fFlushForPrune` from `fDoFullFlush` so that we can skip the code that start at L666 that flush best chain related state. Am I correct?

I wonder why the latter (flush best chain related state) is not needed when running in pruned mode?